src/components/pages/RoleSignup/Applications/Mentee.js

```diff
@@ -49,9 +50,11 @@ const initialFormValues = {
 
 const Mentee = ({ dispatch, error, successPage }) => {
   const { formValues, handleChange } = useForms(initialFormValues);
+  const { push } = useHistory();
 
   const formSubmit = () => {
     dispatch(postNewMenteeAccount(formValues));
+    push('/apply/success');
```

Comment:
a simple technique that could be used prior to pushing the user is to wait a few seconds using a setTimeout to push user ensuring that the 200 response comes back from the userPost.

Reply:
something that occurred us during our PR Critiques session was the idea that this particular dispatch() could be a promise returned. If that be the case we might be able to `thenify` the method and control through the promise return --> routing user to successfully page or setting the error from the server in state and showing that as necessary to the user.

Reply:
https://blog.jscrambler.com/async-dispatch-chaining-with-redux-thunk

You may find a better to do this, but there are a few examples to reflect on here.